src/components/AccountDrawer/MiniPortfolio/Activity/index.tsx

```diff
@@ -82,24 +82,37 @@ const ActivityGroupWrapper = styled(Column)`
   gap: 8px;
 `
 
-function combineActivities(localMap: ActivityMap = {}, remoteMap: ActivityMap = {}): Array<Activity> {
+/* Detects transactions from same account with the same nonce and different hash */
+function wasTxCancelled(localActivity: Activity, remoteMap: ActivityMap, account: string): boolean {
+  // handles locally cached tx's that were stored before we started tracking nonces
+  if (!localActivity.nonce || localActivity.status !== TransactionStatus.Pending) return false
+
+  return Object.values(remoteMap).some((remoteTx) => {
+    if (!remoteTx) return false
```

Comment:
nit: you can remove this line if your `if` statement includes null checks:
`if (remoteTx?.nonce === localActivity.nonce && remoteTx?.receipt?.from.toLowerCase() === account.toLowerCase()) {`

Reply:
I actually prefer carters version better, the breakdown is more clear 😅

Reply:
It might be worth standardizing how we handle potential undefined values. I also noticed the if statement already checks for remoteTx.receipt being undefined instead of returning false earlier. We should try to use the same standard between the two if possible.